docs/usage/models.md

```diff
@@ -495,3 +495,25 @@ _(This script is complete, it should run "as is")_
 
 This is a deliberate decision of *pydantic*, and in general it's the most useful approach. See 
 [here](https://github.com/samuelcolvin/pydantic/issues/578) for a longer discussion on the subject.
+
+
+## Model signature
+
+Any *pydantic* model have its own signature generated based on its fields:
+
+```py
+{!.tmp_examples/models_signature.py!}
+```
+
+Accurate signature is useful for introspection purposes and libraries like `FastAPI` or `hypothesis`.
+
+It also respects your custom defined `__init__` and its arguments:
+
+```py
+{!.tmp_examples/models_signature_custom_init.py!}
+```
+
+To be included in signature, field alias or name must be a valid python identifier. *pydantic* prefer aliases over names, but may use field name if alias is not suitable. 
+If any field's alias and name are both invalid identifiers, `**data` argument will be added.
+
+Also, `**data` argument is always present in signature if `Config.extra` is `Extra.allow`
```

Comment:
```suggestion
Also, `**data` argument is always present in signature if `Config.extra` is `Extra.allow`.

!!! note
    Types in signature are same as declared in model annotations. They are not necessarily the ones that can be parsed. That may change once #1055 solved.
```

Reply:
@samuelcolvin, after this one gets resolved, I guess, we're ready to go 🎉

Reply:
I've rewritten the docs to improve the english, and incorporated this suggestion.